src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/Manifest.cs

```diff
@@ -143,6 +145,16 @@ public static Manifest ReadFrom(Stream stream, Func<string, string> propertyProv
             // Deserialize it
             var manifest = ManifestReader.ReadManifest(document);
 
+            // Update manifest metadata version if version was provided by the CLI command
+            if (propertyProvider is not null && propertyProvider.Target.GetType().Name.Equals("PackArgs"))
```

Comment:
I think I'd rather see a public API change, rather than adding a condition based on a delegate's type name, or other uses of reflection.

Before I spend time looking into the code paths too deeply myself, what other approaches have you considered?  I was wondering if having an additional `Action<Manifest>` delegate (or `Func<Manifest, Manifest>` delegate, so the method signature/public API wouldn't need to change if `Manifest` eventually becomes immutable) would allow this method to provide extensibility after deserialization and before validation.  But looking just a little bit into the code, that would need a bunch of public APIs on PackageBuilder as well, to pass through to here.  I still feel like it might be better than checking the delegate's type name though

Did you try not putting this condition, and always overriding the version when `propertyProvider` has a value for it? What negative side effects are there?

also, the PR description talks about setting the value to 0.0.0, which is no longer happening, so the PR description needs to be updated.

Reply:
> Did you try not putting this condition, and always overriding the version when propertyProvider has a value for it? What negative side effects are there?
Yes, that was my first implementation but realized that some tests were failing. After doing some debugging, I realized that this function is used in multiple scenarios, the one I'm trying to solve is when the customer is using a nuspec.

The tests that were failing is in the [scenario when we are using pack and a .csproj and nuspec](https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs#L6137). That path calls `Manifest.ReadFrom` [using another propertyProvider](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs#L1266) which represents the project file (.csproj) not the nuspec, so the information wasn't correct.

Reply:
I looked at other approach's but given that this is very specific scenario I wanted to do minimal changes.